### ISSUE TITLE
FIREBASE WARNING: set at /devices/thermostats

### ISSUE THREAD

**Issue by maratonat** [KraigM/homebridge-nest#82]
I have been trying to setup the plugin.  Every setup step I followed seemed to match the expected outcome, except after having added the token to the config file I am seeing the following error when attempting to adjust the thermostat temperature:

FIREBASE WARNING: set at /devices/thermostats/7u3hYcaA_NO...75CZ/target_temperature_f failed: Cannot set %s while mode is %s

Any help would be appreciated.

**Comment by maratonat** [comment]
Reinstalled completely and it seems to be working better now. Still don't see nest away in apple Home app.

**Comment by maratonat** [comment]
The issue is back, re-opening.
The problem was alternating between "Cannot set %s while mode is %s" error and "blocked".
The mode can still be changed from Off to Cool, or Heat, or back to Off.  Only changing the temp is problematic.  Also, the concept of Away is not showing up in iOS 10 Home app.

**Comment by DanielAlcorn** [comment]
Yep also having this problem

**Comment by maratonat** [comment]
Apparently this issue crops up when the thermostat is OFF.  When the thermostat is set to heat or cool, the issue seems to go away.  If this is the case, it sounds like Nest is refusing temp changes while thermostat is OFF, reflecting the way the API works.  Can anyone confirm this?

**Comment by DanielAlcorn** [comment]
Yeah I noticed this the other day, can confirm.

**Comment by bensleveritt** [comment]
I'm also seeing this. I'm happy to pitch in for a fix if anyone has any leads?

**Comment by marcgetter** [comment]
This is happening for me on one thermostat but not the other.  I'm not unable to change temperature or mode.  The status is getting read correctly.

**Comment by ericboehs** [comment]
I'm about 99% certain this is because HomeKit is rapid firing the change events to the thermostat. This is fine for a local network (as Apple intended) but Nest rate limits you quickly. I'm not sure if they automatically unblock you or not (still waiting for mine).

The solution here would be to add a debounce to the API calls (reset a ~2 second timer every time a target temp event is received, once the timer expires, send the target temperature to Nest). Perhaps it could be added [here](https://github.com/KraigM/homebridge-nest/blob/master/lib/nest-thermostat-accessory.js#L214), or [here](https://github.com/KraigM/homebridge-nest/blob/f9754a3ac8717f1b7c5c4561cb15f188f5ca025c/lib/nest-connection.js#L122-L125)? 

**Update**: It looks like [Nest recommends](https://developers.nest.com/documentation/cloud/data-rate-limits/) you to "limit requests to one call per minute, maximum".

**Update**: [It seems](http://stackoverflow.com/a/24455529/282720) the rate limit is a sliding one hour.

**Comment by ericboehs** [comment]
@KraigM It seems everyone would be running into this. Do you know of a quick way to debounce the updates?

**Comment by mike-yesware** [comment]
Yep. Same here. I had this problem the very first time I tried to set the temp. More than anything, I'm curious why this isn't a showstopper for all users.

**Comment by clintonb** [comment]
I, too, have this issue. After my initial installation, I moved the temperature slider a lot (not the best UI). This resulted in a lot of API calls being made, and my hitting the (unpublished) rate limit. This is indeed a showstopper for me. I'm happy to help implement/review a solution.

**Comment by igbopie** [comment]
Here it is: 
https://github.com/KraigM/homebridge-nest/pull/115/files

for the impatient:
https://www.npmjs.com/package/homebridge-nest-debounced

**Comment by danbridgland** [comment]
I'm a complete noob, How do I update an existing installation to this version?

**Comment by sirocket** [comment]
I'm with danbridgland too.
